src/capture/capture.cpp

```diff
@@ -290,110 +345,236 @@ static void handle_capture_midi_event(bool pressed)
+		if (ec) {
+			LOG_WARNING("CAPTURE: Cannot open directory '%s': %s",
+			            capture.path.string().c_str(),
+			            ec.message().c_str());
+			return {};
+		}
+		if (!entry.is_regular_file(ec) || entry.path().extension() != ext) {
+			continue;
+		}
+		auto stem = entry.path().stem().string();
+		lowcase(stem);
+		if (starts_with(stem, filename_start)) {
+			const auto index = to_int(strip_prefix(stem, filename_start));
+			highest_index = std::max(highest_index, *index);
+		}
+	}
+	return highest_index;
+}
+
+static void set_next_capture_index(const CaptureType type, int32_t index)
+{
+	switch (type) {
+	case CaptureType::Audio: capture.next_index.audio = index; break;
+	case CaptureType::Midi: capture.next_index.midi = index; break;
+
+	case CaptureType::RawOplStream:
+		capture.next_index.raw_opl_stream = index;
+		break;
+
+	case CaptureType::RadOplInstruments:
+		capture.next_index.rad_opl_instrument = index;
+		break;
+
+	case CaptureType::Video: capture.next_index.video = index; break;
+
+	case CaptureType::RawImage:
+	case CaptureType::UpscaledImage:
+	case CaptureType::RenderedImage:
+		capture.next_index.image = index;
+		break;
+
+	case CaptureType::SerialLog:
+		capture.next_index.serial_log = index;
+		break;
+
+	default: assertm(false, "Unknown CaptureType");
 	}
-#endif
 }
 
-// TODO move raw OPL capture and serial log capture here too
+static bool create_capture_directory()
+{
+	std::error_code ec = {};
+
+	if (!std_fs::exists(capture.path, ec)) {
+		if (!std_fs::create_directory(capture.path, ec)) {
+			LOG_WARNING("CAPTURE: Can't create directory '%s': %s",
+			            capture.path.string().c_str(),
+			            ec.message().c_str());
+			return false;
+		}
+		std_fs::permissions(capture.path, std_fs::perms::owner_all, ec);
+	}
+	return true;
+}
 
-void CAPTURE_Init(Section* sec)
+static void capture_init(Section* sec)
 {
 	assert(sec);
+	const Section_prop* secprop = dynamic_cast<Section_prop*>(sec);
+	if (!secprop) {
+		return;
+	}
 
-	const Section_prop* conf = dynamic_cast<Section_prop*>(sec);
-	assert(conf);
+	Prop_path* capture_path = secprop->Get_path("capture_dir");
+	assert(capture_path);
+
+	// We can safely change the capture output path even if capturing of any
+	// type is in progress.
+	capture.path = std_fs::path(capture_path->realpath);
+	if (capture.path.empty()) {
+		LOG_WARNING("CAPTURE: No value specified for `capture_dir`; defaulting to 'capture' "
+		            "in the current working directory");
+		capture.path = "capture";
+	}
+
+	const std::string prefs = secprop->Get_string("default_image_capture_formats");
 
-	Prop_path* proppath = conf->Get_path("captures");
-	assert(proppath);
+	image_capturer = std::make_unique<ImageCapturer>(prefs);
 
-	capture_dir = proppath->realpath;
+	constexpr auto changeable_at_runtime = true;
+	sec->AddDestroyFunction(&capture_destroy, changeable_at_runtime);
+
+	if (!create_capture_directory()) {
+		return;
+	}
```

Comment:
I was using this PR for routine game and CLI testing too (comparing against other active PRs), and one thing I noticed is that I've got a lot of empty `capture/` directories peppered about my filesystem (well, when ever I launched the executable).

Even for single-line tests, like: `dosbox -c "echo hi" -exit`

Would be really nice to defer this creation until the user requests a capture (then the it's fully justified).

Reply:
Yes, that's capture dir creation regression I referred to above, the one that @FeralChild64 also brought up. Previously I think the capture dir was always created in the DOSBox global config dir, but I switched to using std::fs and that introduced a regression so that the capture dir gets created in the local working dir. I'll fix this and restore the previous behaviour.